Updates Turtle tests to use triple term syntax and reifications

gkellogg opened the pull request:
… instead of quoted triples.

niklasl commented on rdf/rdf12/rdf-turtle/eval/turtle-star-eval-annotation-6.ttl:
```diff
@@ -0,0 +1,3 @@
+PREFIX : <http://example/>
+
+:s :p :o {| :i | :r :z |} .
```
Note that this syntactic form of "naming" the annotation is [problematic](https://lists.w3.org/Archives/Public/public-rdf-star-wg/2024Jan/0097.html) (it won't work in SPARQL, due to the [AlternativePath operator](https://www.w3.org/TR/sparql11-query/#pp-language)).

niklasl replied:
See https://github.com/w3c/rdf-star-wg/issues/116.

afs replied:
Syntax not up to date.

niklasl reviewed (changes requested):
I think these are the required updates for the new reifier naming syntax.

gkellogg commented on rdf/rdf12/rdf-turtle/eval/turtle-star-eval-annotation-8.ttl:
```diff
@@ -0,0 +1,3 @@
+PREFIX : <http://example/>
+
+:s :p :o {| :i | :r :z |} .
```
My branch was out of sync; I had already made these changes. Should be up-to-date now.

gkellogg commented on rdf/rdf12/rdf-turtle/syntax/turtle-star-annotation-6.ttl:
```diff
@@ -0,0 +1,3 @@
+PREFIX : <http://example/>
+
+:s :p :o ~ .
```
Its' a corner-case, as the production for xxx as the production is:

```
reifier               ::= '~' (iri | BlankNode)?
```

Otherwise, `:s :p :o ~` is equivalent to `:s :p :o ~ []` or `:s :p :o {||}` (that last is in the SPARQL grammar, but not the Turtle grammar.

pchampin commented:
This was discussed during the rdf-star meeting on 2024-08-29:
https://www.w3.org/2024/08/29-rdf-star-minutes.html#t04

pchampin commented:
This was discussed during the [#rdf-star meeting on 12 September 2024](https://www.w3.org/2024/09/12-rdf-star-minutes.html#t05).

<details><summary><i>View the transcript</i></summary>

#### [w3c/rdf-tests#135](https://github.com/w3c/rdf-tests/issues/135)
<cite>gkellogg:</cite> first one has been open for a bit. new tests for updated turtle.
  
  

----
</details>

pchampin commented:
This was discussed during the [rdf-star meeting on 12 September 2024](https://www.w3.org/2024/09/12-rdf-star-minutes.html#t05).

<details><summary><i>View the transcript</i></summary>

#### [w3c/rdf-tests#135](https://github.com/w3c/rdf-tests/issues/135)


<cite>gkellogg:</cite> first one has been open for a bit. new tests for updated turtle.
  
  


<cite>AndyS:</cite> about at the stage where I can run those tests.


<cite>ora:</cite> so not to merge yet?


<cite>gkellogg:</cite> yeah, let's wait until someone else is passing the tests.


----
</details>

afs commented:
N-triples tests:

`ntriples-base*` - "base" is a confusing name, especially when copied to `nt-ttl-base`, because they are not about `BASE`.
Suggestions: `ntriples-langdir` named after the terminal or `ntriples-textdir`.

`ntriples-star-bad-reifier-1` tests reifier in the subject position.
`ntriples-star-bad-reifier-2` tests reifier in the object position.
`ntriples-star-bad-reifier-3` tests both subject and object - was a test of the predicate position intended?

`ntriples-star-bad-reifier-4` is unclear. Maybe its testing declaration but it's hard to tell. Simplify to:
`<< <http://example/s> <http://example/p> <http://example/o> >> .`

`ntriples-star-syntax-03` and `ntriples-star-syntax-04` are the same.

`ntriples-star-nested-bad-annotated-syntax-*.nt` don't add much over  `ntriples-star-bnode-bad-annotated-syntax*.nt` because the failure expected is at  the first `{|`.

gkellogg commented:
> N-triples tests:
> 
> `ntriples-base*` - "base" is a confusing name, especially when copied to `nt-ttl-base`, because they are not about `BASE`. Suggestions: `ntriples-langdir` named after the terminal or `ntriples-textdir`.

I renamed these all as you suggested.

> `ntriples-star-bad-reifier-1` tests reifier in the subject position. `ntriples-star-bad-reifier-2` tests reifier in the object position. `ntriples-star-bad-reifier-3` tests both subject and object - was a test of the predicate position intended?
> 
> ``ntriples-star-bad-reifier`-4` is unclear. Maybe it's testing declaration but it's hard to tell. Simplify to: `<< <http://example/s> <http://example/p> <http://example/o> >> .`

I wasn't able to find these tests in the repository; there are likely some additional tests that would be required. Can you point me to them?

> `ntriples-star-syntax-03` and `ntriples-star-syntax-04` are the same.

`ntriples-star-syntax-04` was an old file, and unreferenced; I removed it.

> `ntriples-star-nested-bad-annotated-syntax-*.nt` don't add much over `ntriples-star-bnode-bad-annotated-syntax*.nt` because the failure expected is at the first `{|`.

True, if we were to consolidate, which set should remain?

afs commented:
> I wasn't able to find these tests in the repository; there are likely some additional tests that would be required. Can you point me to them?

`rdf/rdf12/rdf-n-triples/syntax/ntriples-star-bad-reifier-syntax-*.nt`

> > `ntriples-star-nested-bad-annotated-syntax-*.nt` don't add much over `ntriples-star-bnode-bad-annotated-syntax*.nt` because the failure expected is at the first `{|`.
> 
> True, if we were to consolidate, which set should remain?

The  simpler `ntriples-star-bnode-bad-annotated-syntax*.nt`. Annotations aren't in N-triples.

gkellogg commented:
> rdf/rdf12/rdf-n-triples/syntax/ntriples-star-bad-reifier-syntax-*.nt

I didn't see it because the name is slightly different: rdf/rdf12/rdf-n-triples/syntax/ntriples-star-bad-reified-syntax-*.nt. I'll take care of your suggestion.

gkellogg commented:
This will still need updates for further grammar branches and to reconcile https://github.com/w3c/rdf-turtle/pull/70.

TallTed commented on rdf/rdf12/rdf-n-triples/syntax/index.html:
```diff
@@ -538,13 +538,13 @@ <h2>
+            ntriples-langdir-bad-1:
           </a>
-          <span about='../syntax#ntriples-star-nested-bad-annotated-syntax-1' property='mf:name'>N-Triples-star - Bad - annotated triple, nested subject term</span>
+          <span about='../syntax#ntriples-langdir-bad-1' property='mf:name'>N-Triples literal- Bad - undefined base direction</span>
```
```suggestion
          <span about='../syntax#ntriples-langdir-bad-1' property='mf:name'>N-Triples literal — Bad — undefined base direction</span>
```

TallTed commented on rdf/rdf12/rdf-n-triples/syntax/index.html:
```diff
@@ -482,7 +482,7 @@ <h2>
             ntriples-star-bad-reified-4:
           </a>
-          <span about='../syntax#ntriples-star-bad-reified-4' property='mf:name'>N-Triples-star - Bad - reified triples whitespace and terms</span>
+          <span about='../syntax#ntriples-star-bad-reified-4' property='mf:name'>N-Triples-star - Bad - predicate reified triple</span>
```
```suggestion
          <span about='../syntax#ntriples-star-bad-reified-4' property='mf:name'>N-Triples-star — Bad — predicate reified triple</span>
```

TallTed commented on rdf/rdf12/rdf-n-triples/syntax/index.html:
```diff
@@ -554,17 +554,17 @@ <h2>
+            ntriples-langdir-bad-2:
           </a>
-          <span about='../syntax#ntriples-star-nested-bad-annotated-syntax-2' property='mf:name'>N-Triples-star - Bad - annotated triple, nested object term</span>
+          <span about='../syntax#ntriples-langdir-bad-2' property='mf:name'>N-Triples literal- Bad - upper case LTR</span>
```
```suggestion
          <span about='../syntax#ntriples-langdir-bad-2' property='mf:name'>N-Triples literal — Bad — upper case LTR</span>
```

TallTed commented on rdf/rdf12/rdf-n-triples/syntax/manifest.ttl:
```diff
@@ -168,7 +166,7 @@ trs:ntriples-star-bad-reified-3 rdf:type rdft:TestNTriplesNegativeSyntax ;
 
 trs:ntriples-star-bad-reified-4 rdf:type rdft:TestNTriplesNegativeSyntax ;
-   mf:name      "N-Triples-star - Bad - reified triples whitespace and terms" ;
+   mf:name      "N-Triples-star - Bad - predicate reified triple" ;
```
```suggestion
   mf:name      "N-Triples-star — Bad — predicate reified triple" ;
```

afs replied:
This is the way the tests (all three generations) are. Developers use hyphen.
Keep it simple in the manifest because of viewing with the wrong character set.

The rest is automatically produced.

(FYI: Chicago style guide prefers no spaces around em-dashes, which then looks bad in this context.)

TallTed replied:
As I recall, Chicago style guide paradoxically and frustratingly *says* to put no spaces around em-dashes, but *does* put spaces around em-dashes. (Certainly multiple style guides do this.)

TallTed reviewed (changes requested):
separators should be space-emdash-space (or space-double hyphen-space), not space-hyphen-space.
